docs/source/user_guide/using_datasets.rst

```diff
@@ -2335,11 +2348,12 @@ associated metadata.
 
 .. note::
 
-    In most cases, it is recommended to :ref:`add custom attributes as fields
-    directly on<using-labels>` the |Label| object. However, a typical use case
-    for this feature, as opposed to simply storing custom attributes directly
-    on the |Label| object, is to store predictions and associated
-    confidences of a classifier applied to the object patches.
+    In most cases, it is recommended to
+    :ref:`add custom attributes as fields directly on <using-labels>` the
+    |Label| object. However, a typical use case for this feature, as opposed to
```

Comment:
```suggestion
    |Label| object. 
    
    .. code::
    
        sample["label_field"]["custom"attribute"] = 51
    
    However, a typical use case for this feature, as opposed to
```

I'm not sure if we should really keep this section at all, but if so, having a code example of how we recommend they store attributes first could be useful.

Reply:
Good point. I converted this to an actual deprecation warning and included some example code showing what you *should do*